### PR TITLE
Ret 2953 added selectNotification

### PR DESCRIPTION
**Pull request by paul-peasegood**
**Before creating a pull request make sure that:**

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-2953

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java [hunk 1]
```diff
@@ -19,8 +22,9 @@
 
 import java.time.LocalDate;
 import java.util.ArrayList;
-import java.util.List;
+imort java.util.List;
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java:25]: ??

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java [hunk 2]
```diff
@@ -19,8 +22,9 @@
...
 import java.util.Map;
+
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java:27]: Remove this extra line

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java [hunk 3]
```diff
@@ -103,6 +107,25 @@ public void clearSendNotificationFields(CaseData caseData) {
...
+        if (CollectionUtils.isEmpty(sendNotificationTypeItemList)) {
+            return values;
+        }
+        for (SendNotificationTypeItem sendNotificationType : sendNotificationTypeItemList) {
+            String notificationId = sendNotificationType.getId();
+            String label = sendNotificationFormat.format(sendNotificationType);
+            values.add(DynamicValueType.create(notificationId, label));
+        }
+        return values;
+
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/SendNotificationService.java:128]: Missing a closing brace here

**Review by JackReevies — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/RespondNotificationController.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        @RequestBody CCDRequest ccdRequest,
+        @RequestHeader("Authorization") String userToken) {
+
+        if (!verifyTokenService.verifyTokenSignature(userToken)) {
+            log.error(INVALID_TOKEN, userToken);
+            return ResponseEntity.status(FORBIDDEN.value()).build();
+        }
+
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+        // Code here
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/RespondNotificationController.java:102]: Redundant comment

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/RespondNotificationController.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+@RequiredArgsConstructor
+public class RespondNotificationController {
+
+    private static final String INVALID_TOKEN = "Invalid Token {}";
+    private final RespondNotificationService respondNotificationService;
+    private final VerifyTokenService verifyTokenService;
+
+
+    /**
+    * about to start.
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/RespondNotificationController.java:42]: Expand on these javadocs please

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/RespondNotificationController.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        CaseDetails caseDetails = ccdRequest.getCaseDetails();
+        CaseData caseData = caseDetails.getCaseData();
+        respondNotificationService.populateSendNotificationSelection(caseData);
+        return getCallbackRespEntityNoErrors(caseData);
+    }
+
+    /**
+    * About to Submit.
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/RespondNotificationController.java:75]: Expand on these javadocs please

**Review by JackReevies — CHANGES_REQUESTED** [review]

**Review by JackReevies — APPROVED** [review]

**Review by marcjudge — APPROVED** [review]

**Pull request merged**